addAccount

cosmin-195 opened the pull request:
a bit of refactoring

Commit:
commit#1

Commit:
commit#1

cosminflorin1 committed:
Add account done (-validation, -encryption)

cosminflorin1 committed:
DateOfBirth is now a Date

cosminflorin1 committed:
Added encryption -> cyclic dependency

cosminflorin1 committed:
Added encryption

cosminflorin1 committed:
Added encryption

cosminflorin1 committed:
Set username as id for accounts

AnaSireag committed:
[ACCOUNT]: delete account

Commit:
Add account validations

AnaDrambarean committed:
view account details

cosminflorin1 committed:
added get all method

AnaDrambarean committed:
AccountDetails class

Commit:
Account validations

Commit:
validations with annotations

cosminflorin1 committed:
moved Exceptions directory, renamed the repository interface

cosminflorin1 committed:
Merge remote-tracking branch 'origin/anaVultur' into addAccount

# Conflicts:
#	src/main/java/com/bootcamp/demo/model/Account.java

andreibacer commented:
Try to keep packages names consistent, all should be with lower case letters. Also I saw that some of them are defined using singular and some using plural. Can you highlight why?

andreibacer commented on src/main/java/com/bootcamp/demo/Controller/AccountController.java:
```diff
@@ -0,0 +1,29 @@
+        try {
+            service.add(account);
+        }catch(ServiceException exception){
+            throw new ControllerException(exception.getMessage());
```
who is going to handle this exception? An idea might be to encapsulate the result of the operation together with an eventual error code into the response returned.

andreibacer commented on src/main/java/com/bootcamp/demo/data_access/AccountRepository.java:
```diff
@@ -0,0 +1,21 @@
+public class AccountRepository implements AbstractRepository<Account> {
+    public String add(final Account account) throws ExecutionException, InterruptedException {
+        Firestore db = FirestoreClient.getFirestore();
+        DocumentReference docRef = db.collection("accounts").document(account.getUsername());
```
What happens if we are trying to add an account with the same username but with different attributes?

AnaDrambarean committed:
viewAccountDetails

cosmin-195 committed:
renamed packages

cosmin-195 committed:
removed print

cosmin-195 committed:
removed print

cosmin-195 committed:
made adding an account atomic
now also checks if another account with the same user exists

cosmin-195 committed:
add / getAll from Controller now return proper http responses

cosmin-195 committed:
addAccount changes:
- checks if there isn't already an account with the same username
- is now atomic
- returns an appropriate http status along with the result

Commit:
#ValidationService
